### PR TITLE
Fixes #18228: Create an "old-school" crontab technique to manage crontab in /var/spool/cron/crontabs

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/18228

### CODE REVIEW

**Comment by ncharles** [comment]
PR updated with a new commit

File: techniques/system/common/1.0/hooks.cf [hunk 1]
```diff
@@ -737,3 +737,91 @@ bundle agent runhook_cron_process_post_hook(json) {
...
+
+      "na_report"
+        usebundle  => _rudder_common_report_hooks("${reporting[${reportkeys}][technique]}", "result_na", "${reporting[${reportkeys}][id]}", "${reporting[${reportkeys}][name]}", "${reporting[${reportkeys}][value]}", "Restarting service ${cron_service} is not necessary", "${reporting[${reportkeys}][mode]}"),
+        ifvarclass => "!cron_restart.!user_cron_configuration_need_reload_posthook";
+
+      "clean_reporting_context_${reporting[0][id]}"
+        usebundle  => clean_reporting_context;
+
+
+  processes:
```
Comment (amousset) [inline techniques/system/common/1.0/hooks.cf:823]: we have a generic method to check process status using ps.
IIRC cron_bin is part of system techniques and should not be used in normal techniques

Comment (ncharles) [reply techniques/system/common/1.0/hooks.cf:823]: i simply reused what was in cron technique

File: techniques/systemSettings/systemManagement/userCronManagement/1.0/userCronConfiguration.st [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+
+  files:
+      "${cron_base_path}/${cron_user[${cron_index}]}"
+        create        => "true",
+        perms         => mog("600", "${cron_user[${cron_index}]}", "${cron_user[${cron_index}]}"),
```
Comment (ncharles) [inline techniques/systemSettings/systemManagement/userCronManagement/1.0/userCronConfiguration.st:52]: i'm usure of the expected group

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, squash merging this PR

**Pull request merged**